notebooks/name-entity-extraction.ipynb

```diff
@@ -0,0 +1,452 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "output_ner = ner_pipeline(ner_examples[0])\n",
+    "output_ner"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above output displays the results of our model for the first prompt in our examples list. This output is not very intuitive or immediately useful.\n",
+    "Instead, let's see what our model's outputs are if we build a fast and simple gradio app which uses our pipeline to process our outputs on the IPU.\n",
+    "\n",
+    "Using gradio, the `app_for_pipeline` function will build a small app which includes a text prompt and will render the entities which were identified:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gradio as gr\n",
+    "prompt = \"Let's use an app to do some text summarization!\"\n",
+    "out = ner_pipeline(prompt)\n",
+    "\n",
+    "def app_for_pipeline(pipeline, examples=[], description=\"\", label_description=\"\"):\n",
+    "    demo = gr.Blocks(   \n",
+    "        description=description,\n",
+    "    )\n",
+    "    with demo:\n",
+    "        inputs = gr.Textbox(\n",
+    "            lines=3,\n",
+    "        )\n",
+    "        outputs=gr.HighlightedText(\n",
+    "            label=label_description,\n",
+    "            combine_adjacent=True,\n",
+    "            postprocess=True,\n",
+    "            value=dict(text=prompt, entities=out)\n",
+    "        )\n",
+    "        examples_block = gr.Examples(examples=examples, inputs=inputs, outputs=outputs)\n",
+    "        inputs.change(\n",
+    "            fn=lambda x: dict(text=x, entities=pipeline(x)),\n",
+    "            inputs=inputs, outputs=outputs, postprocess=True\n",
+    "        )\n",
+    "    return demo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's see what our examples look like within the app."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "out = ner_pipeline(prompt)\n",
+    "demo = app_for_pipeline(ner_pipeline, ner_examples).launch()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That looks great!\n",
+    "\n",
+    "Using gradio we are able to tell which words are being correctly extracted and identified for different categories using our model.\n",
+    "\n",
+    "This is all aided by the IPU which is quickly processing our inputs and returning the models outputs to create a very responsive interface.\n",
+    "\n",
+    "Next we must detach our model from the IPU to release resources, you can learn more about resource management for the IPU [here](https://github.com/huggingface/optimum-graphcore/blob/main/notebooks/managing_ipu_resources.ipynb)."
```

Comment:
Avoid "here" links; they don't give users any idea what the link is and are bad for accessibility.
```
in our [managing resources notebook](https://github.com/huggingface/optimum-graphcore/blob/main/notebooks/managing_ipu_resources.ipynb).
```